src/lib/template-instance.ts

```diff
@@ -78,34 +79,36 @@ export class TemplateInstance {
           false);
       let node = walker.nextNode();
       // Loop through all the nodes and parts of a template
-      while (partIndex < parts.length && node !== null) {
-        const part = parts[partIndex];
-        // Consecutive Parts may have the same node index, in the case of
-        // multiple bound attributes on an element. So each iteration we either
-        // increment the nodeIndex, if we aren't on a node with a part, or the
-        // partIndex if we are. By not incrementing the nodeIndex when we find a
-        // part, we allow for the next part to be associated with the current
-        // node if neccessasry.
+      while (partIndex < parts.length) {
+        part = parts[partIndex];
         if (!isTemplatePartActive(part)) {
           this._parts.push(undefined);
           partIndex++;
-        } else if (nodeIndex === part.index) {
-          if (part.type === 'node') {
-            const part = this.processor.handleTextExpression(this.options);
-            part.insertAfterNode(node.previousSibling!);
-            this._parts.push(part);
-          } else {
-            this._parts.push(...this.processor.handleAttributeExpressions(
-                node as Element, part.name, part.strings, this.options));
-          }
-          partIndex++;
-        } else {
+          continue;
+        }
+        // Progress the tree walker until we find our next part's node.
+        // Note that multiple parts may share the same node (attribute parts
+        // on a single element), so this loop may not run at all.
+        while (nodeIndex < part.index) {
           nodeIndex++;
-          if (node.nodeName === 'TEMPLATE') {
-            _prepareInstance((node as HTMLTemplateElement).content);
+          if (node!.nodeName === 'TEMPLATE') {
+            _prepareInstance((node as HTMLTemplateElement).content)!;
           }
-          node = walker.nextNode();
+          if (!(node = walker.nextNode())) {
```

Comment:
```suggestion
          if ((node = walker.nextNode()) !=== null) {
```

I prefer direct comparisons rather than boolean conversions.